app/views.py

```diff
@@ -8,3 +8,29 @@
 @App.route('/')
 def public_timeline():
     return render_template('public_timeline.html', flashes=db_utils.get_all_twittes())
+
+@app.route('/register', methods=['GET', 'POST']) # akef
+def register():
+    """Registers the user."""
+    if g.user:
+        return redirect(url_for('timeline'))
+    error = None
+    if request.method == 'POST':
+        if not request.form['username']:
+            error = 'You have to enter a username'
+        elif not request.form['email'] or \
+                '@' not in request.form['email']:
+            error = 'You have to enter a valid email address'
+        elif not request.form['password']:
+            error = 'You have to enter a password'
+        elif request.form['password'] != request.form['password2']:
+            error = 'The two passwords do not match'
+        elif get_user_id(request.form['username']) is not None:
+            error = 'The username is already taken'
+        else:
+            db = get_db()
```

Comment:
ازاي بس  دي حجات زيادة للتأكيد عشان الإيررورز وكده بحيث يعرف هو غلطان فين
مش مضرة يعني

On Sun, Aug 14, 2016 at 12:07 AM, Ahmed Maher <notifications@github.com>
wrote:

> In app/views.py
> <https://github.com/CATWebTeam/git-workshop/pull/5#discussion_r74692801>:
>
> > +        return redirect(url_for('timeline'))
> > +    error = None
> > +    if request.method == 'POST':
> > +        if not request.form['username']:
> > +            error = 'You have to enter a username'
> > +        elif not request.form['email'] or \
> > +                '@' not in request.form['email']:
> > +            error = 'You have to enter a valid email address'
> > +        elif not request.form['password']:
> > +            error = 'You have to enter a password'
> > +        elif request.form['password'] != request.form['password2']:
> > +            error = 'The two passwords do not match'
> > +        elif get_user_id(request.form['username']) is not None:
> > +            error = 'The username is already taken'
> > +        else:
> > +            db = get_db()
>
> you don't need this
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CATWebTeam/git-workshop/pull/5/files/91463a8b79ad1996ef7c7f3c0e41010ccbf6565a#r74692801>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMhI18uAKLlDSR8_ZOk39lEIZv6uzXhHks5qfkAUgaJpZM4Jjw-i>
> .
>